add draft related docs (#198)
bufbuild/docs.buf.build: docs/bsr/overview.md

  - buf.build/acme/units
```

> Although we **do not recommend** it, in some situations you may need to pin a module to a specific version, you can do so by specifying tag, commit, or draft in your `deps` after the `:` delimiter. Ideally, authors keeps modules backwards-compatible and avoid breaking changes so you can *always* rely on the latest version.

Reviewer: Is this correct, i.e., can we specify a dependency by its draft name?

```
deps:
  - buf.build/acme/paymentapis:dev-branch
```

Iirc this will error:

> ... a module with a dependency pinned to a draft commit is not allowed

Author: yeh we changed the behaviour to allow using draft as dependency, but disallow it on push

Reviewer: I see, so drafts work everywhere _except_ for the push command?

Author: yeh basically it just works like all other references except you cannot push a module with a draft dependency

Each module on the BSR exists as a snapshot, and contains a unique reference associated with every change.

A reference is a way to refer to a single version of the repository. While a reference always _resolves_ to a single snapshot of the repository, it can be either a commit, a tag, or a draft.

**Commit**: Every push of new content to a repository is associated with a commit that identifies that change in the schema. The commit is created after a successful push. This means that unlike Git, the commit only exists on the BSR repository and not locally.

**Tag**: A reference to a single commit but with a human readable name, similar to a Git tag. It is useful for identifying commonly referenced commits&mdash;like a release.

> **Draft**: A temporary commit in development workflow with a human readable name, similar to a Git feature branch but without history. It is useful for developing on the module and want to commit outside the main branch. Can be replaced by another commit with the same draft name and is deletable.

Reviewer: Maybe worth adding a note that drafts cannot be used for dependency resolution?

Reviewer: Ye, I think we need some blurb (probably in this section) that explicitly says modules with dependencies referencing a draft cannot be pushed to the BSR. 

Tbh I came away a bit confused because this wasn't immediately obvious to me (unless I missed it).

> **Draft**: A temporary commit in development workflow with a human readable name, similar to a Git feature branch but without history. It is useful for developing on the module and want to commit outside the main branch. Can be replaced by another commit with the same draft name and is deletable. When it is used as a dependency in `buf.yaml` of a module, the module cannot be pushed until you updated to a non-draft commit of the dependency.

Reviewer: ```suggestion
**Draft**: A temporary commit in development workflow with a human readable name, similar to a Git feature branch but without history. It is useful for iterating on a module while keeping those changes outside the main branch. Can be replaced by another commit with the same draft name and is deletable. When it is used as a dependency in `buf.yaml` of a module, the module cannot be pushed until you update to a non-draft commit of the dependency.
```
